cloud/aws/aws_cloud_provider.go

```diff
@@ -95,7 +95,7 @@ func (sp *AwsScalingProvider) scaleOut(workerPool *structs.WorkerPool) error {
 	// and availability zones.
 	availabilityZones := asg.AutoScalingGroups[0].AvailabilityZones
 	terminationPolicies := asg.AutoScalingGroups[0].TerminationPolicies
-	newCapacity := *asg.AutoScalingGroups[0].DesiredCapacity + int64(1)
+	newCapacity := *asg.AutoScalingGroups[0].DesiredCapacity + int64(workerPool.ScaleFactor)
```

Comment:
The scalefactor + desiredcapacity should be checked against the ASG's max value, using the greater of the two.

Reply:
Changes applied, thanks!